modules/lib/generators.nix

```
@@ -99,4 +99,31 @@
     in attrs: ''
       ${concatStringsSep "\n" (mapAttrsToList convertAttributeToKDL attrs)}
     '';
+
+  toSCFG = let
+    inherit (lib)
+      concatStringsSep filterAttrs isAttrs isBool isFloat isInt isList isString
+      mapAttrsToList;
+    inherit (builtins) typeOf toJSON;
+
+    toSCFG' = v:
+      if isAttrs v then ''
+        {
+          ${toSCFG v}
+        }
+      '' else if isList v then
+        concatStringsSep " " v
+      else if isString v then
+        v
+      else if isInt v || isFloat v || isBool v then
+        toJSON v
+      else
+        abort "toSCFG: type ${typeOf v} is unsupported";
+
+    filterSCFG = filterAttrs (_: val: !isNull val && val != { });
+
+    sectionSCFG = mapAttrsToList (key: val: "${key} ${toSCFG' val}");
+
+    toSCFG = v: concatStringsSep "\n" (sectionSCFG (filterSCFG v));
```

```suggestion
    toSCFG = v: concatLines (sectionSCFG (filterSCFG v));
```

(This also takes care of adding a final newline)

---

Adds a blank line at the end of a block due to the mutual recursion. Looks fine to me, but can reformat if needed.